libraries/AP_Beacon/AP_Beacon_Nooploop.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+#include "AP_Beacon_Nooploop.h"
+#include <ctype.h>
+#include <stdio.h>
+
+#define NOOPLOOP_INVALID_VAL -8388000
```

Comment:
This value needs explaining.  Is it a magic bit pattern?  Something directly copied from a datasheet?

Reply:
Yes, it is copied from a datasheet. I will add a comment about this.